### ISSUE TITLE
Google Maps & Flutter - Heat Maps support for Flutter Web

### ISSUE THREAD

**Issue by tparadinas** [flutter/flutter#86811]
Currently there is no support/availabilty to use heat maps functionality for Flutter Web, but is perfectly working for Flutter APP.

We are developing a GIS full application with Flutter using both, APP and Web, with the support of @domesticmouse. Heat maps are something essential to show specific data layers for that project.

When are heat maps expected to be supported for Flutter Web?  Any ideas or alternatives meanwhile?

Thank you so much in advance!

**Comment by domesticmouse** [comment]
Hey @ditman, any ideas on how this can be implemented?

**Comment by ditman** [comment]
Quick question; **how are you rendering Heatmaps in the Google Maps Flutter plugin for mobile currently?** I've checked the platform_interface and I don't see any methods there that have to do with heatmaps?

If it's a homegrown solution, well, first that would need to be merged to the plugin platform_interface first, so there's an unified API that can be implemented both for mobile and web.

With regards to the web implementation, I think most of the pieces are ready to support Heat Maps:

* It's supported by the JS-interop layer: [package:google_maps/visualization](https://pub.dev/documentation/google_maps/latest/google_maps_visualization/HeatmapLayer-class.html)
* In the index.html, when including the JS SDK the `visualization` library needs to be added [see this](https://developers.google.com/maps/documentation/javascript/earthquakes#html).

There might be some questions about the data formatting/conversion between the mobile and web versions of the maps SDK, but once those are eased, this should be doable.

**Comment by tparadinas** [comment]
Yes, it is a third-party plugin that we found...

Glad to hear that most of the pieces are ready to support Heat Maps. I hope it can be used soon!

Thank you very much for your answer @ditman!

**Comment by ditman** [comment]
Blocked by: https://github.com/flutter/flutter/issues/33586

**Comment by Rexios80** [comment]
I have basic heatmap support for web working in https://github.com/flutter/plugins/pull/5274. iOS implementation and testing needs done though.

**Comment by ditman** [comment]
@Rexios80 yes! I really want to start landing your PRs for google maps flutter, I've been busy with other framework stuff, but we'll get there!

**Comment by wangela** [comment]
Heatmap layer support has been released for apps targeting Android, iOS, and web in google_maps_flutter [v2.8.0](https://pub.dev/packages/google_maps_flutter/changelog#280) 🚀
